src/msgspec/_core.c

```diff
@@ -7938,6 +7939,66 @@ Struct_copy(PyObject *self, PyObject *args)
     return NULL;
 }
 
+
+static PyObject* get_deepcopy_func() {
```

Comment:
Trying a new style of handling these kinds of imports here. We initialize the field on the module state as `NULL`, and import on an as-needed basis. I think this is a fairly reasonable compromise between performance and complexity. After the first import, it's just one additional `x == NULL` check, which should be negligible in terms of performance. 

The only real downside I can see is that you now have got to remember to use the "getter function" and cannot rely on the module state.

@ofek lmk what you think about this.

Reply:
I agree that we should not lazily import this module however I think your implementation would be extremely useful for basically everything else, if you wouldn't mind trying or documenting the pattern somewhere for a future PR?